sw/device/lib/dif/dif_sram_ctrl.h

```diff
@@ -11,256 +11,275 @@
+   * scrambling key is still ephemeral (i.e., it is derived using entropy from
+   * CSRNG), but a default all-zero value is used as the key seed. This could
+   * happen when the scrambling key seeds have not yet been provisioned to OTP.
+   */
+  kDifSramCtrlStatusScrKeySeedValid,
+  /**
+   * Hardware initialization triggered via `dif_sram_ctrl_scramble` or
+   * `dif_sram_ctrl_wipe` has completed.
+   */
+  kDifSramCtrlStatusInitDone,
+} dif_sram_ctrl_status_t;
 
 /**
- * A toggle state: enabled, or disabled.
- *
- * This enum may be used instead of a `bool` when describing an enabled/disabled
- * state.
+ * SRAM Controller lockable functionality.
  */
-typedef enum dif_sram_ctrl_toggle {
+typedef enum dif_sram_ctrl_lock {
   /**
-   * The "enabled" state.
+   * SRAM scrambling key renewal and "wiping" lock, which includes the following
+   * API: `dif_sram_ctrl_scramble`, `dif_sram_ctrl_request_new_key`
+   * and `dif_sram_ctrl_wipe`.
    */
-  kDifSramCtrlToggleEnabled,
+  kDifSramCtrlLockCtrl = 0,
   /**
-   * The "disabled" state.
+   * Code execution from SRAM lock, which includes the following API:
+   * `dif_sram_ctrl_exec_set_enabled`.
    */
-  kDifSramCtrlToggleDisabled,
-} dif_sram_ctrl_toggle_t;
+  kDifSramCtrlLockExec,
+} dif_sram_ctrl_lock_t;
+
+/**
+ * SRAM Controller DIF result codes.
+ *
+ * These codes represent an extension to the common `dif_result_t`, and are
+ * intended to be used in the DIF operations that may result in these extra
+ * result codes. For the rest, the common `dif_result_t` should be used.
+ *
+ * Due to these extended error codes being SRAM Controller DIF wide, and not
+ * on per-function basis, extra care is required by the DIF implementer, to
+ * ensure that a meaningful error code for a particular operation is returned.
+ */
+typedef enum dif_sram_ctrl_result {
+  kDifSramCtrlOk = kDifOk,
+  kDifSramCtrlError = kDifError,
+  kDifSramCtrlBadArg = kDifBadArg,
+  kDifSramCtrlLocked = kDifLocked,
+  /**
+   * SRAM Controller scrambling operation is in progress.
+   */
+  kDifSramCtrlScramblingBusy,
+} dif_sram_ctrl_result_t;
 
 /**
  * Creates a new handle for SRAM Controller.
  *
  * This function does not actuate the hardware.
  *
- * @param params Hardware instantiation parameters.
- * @param[out] handle Out param for the initialized handle.
+ * @param base_addr The MMIO base address of the IP.
+ * @param[out] sram_ctrl Out param for the initialized handle.
  * @return The result of the operation.
  */
 OT_WARN_UNUSED_RESULT
-dif_sram_ctrl_result_t dif_sram_ctrl_init(dif_sram_ctrl_params_t params,
-                                          dif_sram_ctrl_t *handle);
+dif_result_t dif_sram_ctrl_init(mmio_region_t base_addr,
+                                dif_sram_ctrl_t *sram_ctrl);
 
 /**
- * Trigger one alert event.
+ * Performs SRAM scrambling.
+ *
+ * This is a compound operation consisting of "scrambling" or in other words
+ * logical re-mapping of the physical addresses. Followed by entire SRAM
```

Comment:
nit: scrambling covers both data and address scrambling. This description makes it sound as if it was just address scrambling.

Reply:
Done, but please check if this conveys the meaning better.